sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareDirectoryStorageResourceContainer.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Azure.Storage.Files.Shares;
+
+namespace Azure.Storage.DataMovement.Files.Shares
+{
+    internal class ShareDirectoryStorageResourceContainer : StorageResourceContainer
```

Comment:
I know we discussed this before, but I forget where we landed. Can we combine this and `ShareStorageResourceContainer` and just have some option that defines the path inside the share if needed? Like we did with Blob Container and a virtual directory.

Reply:
I'd actually be interested to quickly revisit that conversation, as it was made when these types were public rather than internal implementation details. I also think the existence of actual directories with different base clients compared to containers and prefixes weighs in.

Reply:
I suppose also it would depend on how much code duplication occurs. The code was nearly identical between container and "virtual directory." I imagine shares vs directories would be quite a bit different, but I suppose that will need to be looked into.

Reply:
Yeah, you may be right about there not being much code duplication between shares and directories since they theoretically use different clients. Though perhaps "Share" level can just root directory. I think we discussed that we didn't want to get into the business of creating shares since that comes with a lot of extra options and that may make the above statement possible. Happy to discuss further.